FilePersistence/src/version_control/Commit.cpp

```diff
@@ -103,56 +103,86 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 		return false;
 }
 
-QStringList Commit::nodeLinesFromId(Model::NodeIdType id, bool findChildrenByParentId) const
+bool Commit::isValidMatch(QString content, int indexOfId, int& start, int& end, bool findChildrenByParentId) const
 {
-	auto idtext =  id.toString();
-	QStringList matches;
-	for (auto file : files())
+	start = indexOfId;
+	end = indexOfId;
+
+	// start is the first character of the line containing id
+	while (start >= 0 && content[start] != '\n')
 	{
-		int indexOfId = 0;
-		auto content = QString::fromUtf8(file->content());
+		// String is of the form {.*} {id}
+		if (findChildrenByParentId == 0 && content[start] == '}')	return 0;
+		start--;
+	}
+	start++;
 
-		indexOfId = content.indexOf(idtext, indexOfId);
-		if (indexOfId != -1)
-		{
-			bool invalid = false;
-			auto start = indexOfId;
-			auto end = indexOfId;
+	// end is the character after the line containing id
+	while (end <= content.size() && content[end] != '\n')
+	{
+		// String is of the form {id} {*.}
+		if (findChildrenByParentId == 1 && content[end] == '{')	return 0;
+		end++;
+	}
+	return 1;
+}
 
-			// start is the first character of the line containing id
-			while (start >= 0 && content[start] != '\n')
-			{
-				// String is of the form {.*} {id}
-				if (!findChildrenByParentId)
-					if (content[start] == '}')
+QStringList Commit::nodeLinesFromId(Model::NodeIdType Id, QString workDir_, bool isWorkingDir,
```

Comment:
We do not need two additional parameters connected to working directory. One string should be enough. If the string is null (`QString.isNull()`) then it's not a working directory, otherwise it is.